<launch>

    <!-- Arguments for map and RViz -->
    <arg name="pub_map"     default="true"/>
    <arg name="rviz"        default="false"/>

    <!-- Arguments for cameras_tf launch -->
    <arg name="navigation"          default="false"/>
    <arg name="science"             default="false"/>
    <arg name="maintenance"         default="false"/>

    <arg name="navigation_rviz"          default="false"/>
    <arg name="science_rviz"             default="false"/>
    <arg name="maintenance_rviz"         default="false"/>

    <arg name="d435"    default="true"/>
    <arg name="d435i"   default="true"/>
    <arg name="t265"    default="true"/>
    <arg name="log_1"   default="true"/>
    <arg name="log_2"   default="true"/>
    <arg name="zed"     default="true"/>

    <!-- Include launches for tfs -->
    <include if="$(arg navigation)" file="$(find sirio_utilities)/launch/cameras_tf/navigation_tf.launch">
        <arg name="rviz" value="$(arg navigation_rviz)"/>
    </include>
    <include if="$(arg science)" file="$(find sirio_utilities)/launch/cameras_tf/science_tf.launch">
        <arg name="rviz" value="$(arg science_rviz)"/>
    </include>
    <include if="$(arg maintenance)" file="$(find sirio_utilities)/launch/cameras_tf/maintenance_tf.launch">
        <arg name="rviz" value="$(arg maintenance_rviz)"/>
    </include>

    <!-- Args for device names -->

    <arg name="log_1_device_name"          default="/dev/video8"/>
    <!-- <arg name="log_2_device_name"          default="/dev/video16"/> -->
    <arg name="zed_device_name"            default="/dev/video0"/>

    <!-- Include the launch file for the D435i camera -->
    <include if="$(arg d435i)" file="$(find sirio_utilities)/launch/cameras/d435i.launch">
        <arg name="model"   value="false"/>
        <arg name="pub_tf"  value="false"/>    
    </include>

    <!-- Include the launch file for the D435 camera -->
    <include if="$(arg d435)" file="$(find sirio_utilities)/launch/cameras/d435.launch">
        <arg name="model"   value="false"/>
        <arg name="pub_tf"  value="false"/>    
    </include>

    <!-- Include the launch file for the T265 camera -->
    <!-- <include if="$(arg t265)" file="$(find sirio_utilities)/launch/cameras/t265.launch">
        <arg name="pub_tf" value="false"/>    
    </include> -->

    <!-- Include the launch file for the usb camera -->
    <include if="$(arg log_1)" file="$(find sirio_utilities)/launch/cameras/usb_cam.launch">
        <arg name="pub_tf"          value="false"/> 
        <arg name="video_device"    value="$(arg log_1_device_name)"/> 
        <arg name="node_name"       value="log_1"/> 
    </include>

    <!-- Include the launch file for the usb camera -->
    <!-- <include if="$(arg log_2)" file="$(find sirio_utilities)/launch/cameras/usb_cam.launch">
        <arg name="pub_tf"          value="false"/> 
        <arg name="video_device"    value="$(arg log_2_device_name)"/> 
        <arg name="node_name"       value="log_2"/> 
    </include> -->

    <!-- Include the launch file for the zed camera -->
    <include if="$(arg zed)" file="$(find zed_cpu_ros)/launch/zed_cpu_ros.launch">
        <arg name="pub_tf"          value="false"/> 
        <arg name="device_name"     value="$(arg zed_device_name)"/> <!-- Make sure the right video# is written  -->
    </include>

    <!-- Launch save image service -->
    <node name="camera_image_saver" pkg="sirio_utilities" type="camera_image_saver.py"/>
    
    <!-- Launch encoder -->
    <!-- <node name="$(anon encoder)" pkg="sirio_utilities" type="encode_camera.py"/> -->
    
    <!-- Launch RViz -->
    <node if="$(arg rviz)" name="$(anon rviz)" pkg="rviz" type="rviz" args="-d $(find sirio_utilities)/config/rviz/cameras.rviz" required="true" />
 
</launch>
